dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -325,6 +325,28 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
   def makeUDF[A1: TypedEncoder, A2: TypedEncoder, A3: TypedEncoder, A4: TypedEncoder, A5: TypedEncoder, R: TypedEncoder](f: (A1, A2, A3, A4, A5) => R):
   (TypedColumn[T, A1], TypedColumn[T, A2], TypedColumn[T, A3], TypedColumn[T, A4], TypedColumn[T, A5]) => TypedColumn[T, R] = functions.udf(f)
 
+
+  /** Prepends a new column to the Dataset.
+    * {{{
+    *   case class X(i: Int, j: Int)
+    *   val f: TypedDataset[X] = TypedDataset.create(X(1,1) :: X(1,1) :: X(1,10) :: Nil)
+    *   val fNew: TypedDataset[(Int,Int,Boolean)] = f.withColumn(f('j) === 10)
+    * }}}
+    */
+  def withColumn[A: TypedEncoder, H <: HList, FH <: HList, Out](ca: TypedColumn[T, A])(
+    implicit
+    g: Generic.Aux[T, H],
+    init: Prepend.Aux[H, A :: HNil, FH],
+    t: Tupler.Aux[FH, Out],
```

Comment:
I'm not entirely convinced by the use of `Tupler` here. This is throwing away all the names written by the user so far, which wouldn't really be faithful to the Spark counterpart...